### PR TITLE
Explain spherical soma and connectedness.

### LINKED ISSUE #2161: Bug? num_branches is something unexpected.
Let's take a cell morph in `swc` format:

```
##n,type,x,y,z,radius,parent
1 1 0.0        0.0   0.0 6.30785 -1
```

Arbor, how many branches is this, and can you print the `segment_tree`?

```
python -c 'import arbor;swc=arbor.load_swc_neuron("soma_branch.swc");print(swc.num_branches);print(swc.to_segment_tree())'
1
<arbor.segment_tree:
(segment_tree (
  (segment 0 (point -6.30785 0 0 6.30785) (point 0 0 0 6.30785) 1)
  (segment 1 (point 0 0 0 6.30785) (point 6.30785 0 0 6.30785) 1))
  (npos 0))>
```

Thanks buddy, that's exactly what I thought, the single compartment soma, defacto interpreted as a sphere, is converted to two cylinders with a surface equal to that of the sphere. [As per docs.](https://docs.arbor-sim.org/en/latest/fileformat/swc.html)

Let's add a dendrite compartment to the `swc`, and connect it to the soma:

```
##n,type,x,y,z,radius,parent
1 1 0.0        0.0   0.0 6.30785 -1
2 3 6.30785    0.0   0.0 1.5      1
```

Arbor?

```
python -c 'import arbor;swc=arbor.load_swc_neuron("soma_branch.swc");print(swc.num_branches);print(swc.to_segment_tree())'
3
<arbor.segment_tree:
(segment_tree (
  (segment 0 (point -6.30785 0 0 6.30785) (point 0 0 0 6.30785) 1)
  (segment 1 (point 0 0 0 6.30785) (point 6.30785 0 0 6.30785) 1)
  (segment 2 (point 0 0 0 1.5) (point 6.30785 0 0 1.5) 3))
  (npos 0 0))>
```

Hwhat? One segment is added, sure, but there are 3 branches now? On a perfectly linear segment tree? And `(npos 0)` is now `(npos 0 0)`? And the dendrite segment is placed at `0,0,0`. Who knows what's going on? I would expect `num_branches` to be 1, _maybe_ 2, but not 3.

Let's add a third compartment:

```
##n,type,x,y,z,radius,parent
1 1 0.0        0.0   0.0 6.30785 -1
2 3 6.30785    0.0   0.0 1.5      1
3 3 306.30785  0.0   0.0 1.5      2
```

Arbor, you know the drill.

```
python -c 'import arbor;swc=arbor.load_swc_neuron("soma_branch.swc");print(swc.num_branches);print(swc.to_segment_tree())'
3
<arbor.segment_tree:
(segment_tree (
  (segment 0 (point -6.30785 0 0 6.30785) (point 0 0 0 6.30785) 1)
  (segment 1 (point 0 0 0 6.30785) (point 6.30785 0 0 6.30785) 1)
  (segment 2 (point 6.30785 0 0 1.5) (point 306.308 0 0 1.5) 3))
  (npos 0 0))>
```

Hmm, now the dendrite no longer starts at the origin, but the coordinate of the first dendritic compartment. Sure, I can see how that might have caused a second branch in the previous example, but surely now the number of branches should be 1?

### PR DESCRIPTION
**Pull request by thorstenhater**
Extend docs to cover two topics I recently interact with. These seemed confusing to me,
so I expect more people will feel this way.

- Spherical soma in SWC and its consequence for attaching dendrites.
- Connectedness of components.

Closes #2161

### CODE REVIEW

File: doc/concepts/morphology.rst [hunk 1]
```diff
@@ -393,8 +393,43 @@ multiple soma and dendrite segments in branch 0.
...
+
+Two components are connected -- and thus one component -- if there are points in
+both that are connected. In short, a component is defined as the maximal set of
+connected points.
+
+.. figure:: ../gen-images/label_branch.svg
+  :width: 800
+  :align: center
+
+This mean a simple fork -- as that between branches 2, 3, and 4 in the image
```
Comment (AdhocMan) [inline doc/concepts/morphology.rst:414]:
```suggestion
This means a simple fork -- as that between branches 2, 3, and 4 in the image
```

File: doc/fileformat/swc.rst [hunk 2]
```diff
@@ -114,16 +114,28 @@ joined at the soma center. It differs in two ways:
 
 * The soma is extended along the x-axis, not the y-axis.
 * The soma is constructed from three points, the first at ``x=x0-r``, the second with
-  ``x=x0`` and the third at ``x=x0+r``, to form a single section, with all dendrites, axons
-  and apical dendrites attached to the center of the soma with "zero resistance wires".
+  ``x=x0`` and the third at ``x=x0+r``, to form a single section.
+* All dendrites, axons, and apical dendrites attached to the center of the soma with "zero resistance wires".
```
Comment (AdhocMan) [inline doc/fileformat/swc.rst:118]:
```suggestion
* All dendrites, axons, and apical dendrites are attached to the center of the soma with "zero resistance wires".
```

File: doc/scripts/gen-labels.py [hunk 3]
```diff
@@ -343,6 +345,7 @@ def write_morphology(name, morph):
 
 f.write("\n############# locsets (label_morph)\n\n")
 for label in locsets:
+    print(label)
```
Comment (AdhocMan) [inline doc/scripts/gen-labels.py:348]: Is that a leftover?

**Review by AdhocMan — CHANGES_REQUESTED** [review]

File: doc/scripts/gen-labels.py (hunk 3 above)
Comment (thorstenhater) [reply doc/scripts/gen-labels.py]: Yes.

**Pull request merged**